src/storage/conversion/ErrorToJsonConverter.ts

```diff
@@ -0,0 +1,53 @@
+import arrayifyStream from 'arrayify-stream';
+import { BasicRepresentation } from '../../ldp/representation/BasicRepresentation';
+import type { Representation } from '../../ldp/representation/Representation';
+import { APPLICATION_JSON, INTERNAL_ERROR } from '../../util/ContentTypes';
+import { HttpError } from '../../util/errors/HttpError';
+import { InternalServerError } from '../../util/errors/InternalServerError';
+import type { RepresentationConverterArgs } from './RepresentationConverter';
+import { TypedRepresentationConverter } from './TypedRepresentationConverter';
+
+/**
+ * Converts an Error object to JSON by copying its fields.
+ */
+export class ErrorToJsonConverter extends TypedRepresentationConverter {
+  public constructor() {
+    super(INTERNAL_ERROR, APPLICATION_JSON);
+  }
+
+  public async handle({ representation }: RepresentationConverterArgs): Promise<Representation> {
+    const errors = await arrayifyStream(representation.data);
+    if (errors.length !== 1) {
+      throw new InternalServerError('Only single errors are supported.');
+    }
+    const error = errors[0] as Error;
```

Comment:
I have seem this pattern before. We might want a `getSingleItem` helper in our stream helpers.